azureDatabaseUtils/src/test/java/bio/terra/workspace/azureDatabaseUtils/create/CreateDatabaseDaoTest.java

```diff
@@ -0,0 +1,87 @@
+package bio.terra.workspace.azureDatabaseUtils.create;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.workspace.azureDatabaseUtils.BaseUnitTest;
+import java.sql.Types;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+public class CreateDatabaseDaoTest extends BaseUnitTest {
+
+  @Autowired private CreateDatabaseDao createDatabaseDao;
+
+  @Autowired private JdbcTemplate jdbcTemplate;
+
+  private final String testDatabaseName = "testCreateDatabase";
+  private final String testRoleName = "testCreateRole";
+
+  @BeforeEach
+  @AfterEach
+  void cleanup() {
+    jdbcTemplate.execute("DROP DATABASE IF EXISTS " + testDatabaseName);
+    jdbcTemplate.execute("DROP ROLE IF EXISTS \"%s\"".formatted(testRoleName));
+  }
+
+  @Test
+  void testCreateDatabase() {
+    createDatabaseDao.createDatabase(testDatabaseName);
+
+    jdbcTemplate
+        .query(
+            "select count(*) from pg_database where datname = lower(?)",
+            new Object[] {testDatabaseName},
+            new int[] {Types.VARCHAR},
+            (rs, rowNum) -> rs.getInt(1))
+        .forEach(count -> assertThat(count, equalTo(1)));
+  }
+
+  @Test
+  void testCreateRole() {
+    final String testUserOID = "test";
+    createRoleFunction();
+    assertThat(createDatabaseDao.createRole(testRoleName, testUserOID), equalTo("created"));
+    assertThat(createDatabaseDao.createRole(testRoleName, testUserOID), equalTo("exists"));
+  }
+
+  @Test
+  void testGrantAllPrivileges() {
+    jdbcTemplate.execute("CREATE DATABASE " + testDatabaseName);
```

Comment:
Why not call `createDatabaseDao.createDatabase(testDatabaseName)`?

Reply:
I was trying to have this test be independent of the createDatabase call